cpp/src/groupby/common/aggregation_requests.hpp

```diff
@@ -52,6 +53,33 @@ using AggRequestType = std::pair<gdf_column*, operators>;
  *---------------------------------------------------------------------------**/
 using SimpleAggRequestCounter = std::pair<AggRequestType, gdf_size_type>;
 
+
+static constexpr std::array<operators, 4> simple_aggregations = {SUM, MIN, MAX, COUNT};
+
+static constexpr std::array<operators, 2> complex_aggregations = {MEDIAN, QUANTILE};
+
+
+/**---------------------------------------------------------------------------*
+ * @brief  To verify that the input operator is part of simple_aggregations list.
+ * Note that in this kind of aggregators can be computed in a single pass scan.
+ * In the other hand compound aggregation like AVG need to be computed by simple
+ * ones (SUM and COUNT).
+ *---------------------------------------------------------------------------**/
+inline bool is_simple(operators op) {
+  return std::any_of(simple_aggregations.begin(), simple_aggregations.end(),
```

Comment:
I think `find` would be clearer: 
```suggestion
  return std::find(simple_aggregations.begin(), simple_aggregations.end(), op) != simple_aggregations.end();
```

Reply:
Also, if you replace the `std::array` above with `std::set` you won't need to hard-code the size, and can just use 
```
return simple_aggregations.find(op) != simple_aggregations.end();
```

(C++20 adds set::contains() which would simplify it more).

What is the reason for using `std::array`? I don't think it will make the search that much faster.  If you want it to be faster, maybe just assign values to the enum such that all aggregations < some constant are simple, and all others are complex...

Reply:
The purpose of making it an `array` means that you can make `is_simple/is_complex` to be `constexpr`.

Reply:
Just to be clear. Due to there is not any functions like `any_of` of `find` as `constexpr` function in C++14. Is still convenient to use `std::array`? or do you know some function as `constexpr`?

Reply:
Here's a constexpr version:
```
template <class T, size_t N>
constexpr bool contains(std::array<T,N> const& haystack, T needle){
    for(auto i = 0u; i < N; ++i){
       if(haystack[i] == needle)
           return true;
    }
    return false;
}
```
Example: https://wandbox.org/permlink/tkAJVkAwTbG5I7pb

Reply:
@jrhemstad why are you concerned with this being constexpr? It's a host function, never run on the device. I doubt it will be a bottleneck, so it seems like early optimization to worry about that. Also, using `std::array` requires hard-coding the array size, which is inconvenient and could lead to bugs.

Reply:
It has nothing to do with performance. It's so that you could template specialize on this function.

Reply:
All right!